identity/service.go

```diff
@@ -69,64 +75,80 @@ type Storage struct {
 // CreateIdentity will register a new SkipChain and add it to our list of
 // managed identities.
 func (s *Service) CreateIdentity(ai *CreateIdentity) (network.Message, onet.ClientError) {
-	log.Lvlf3("%s Creating new identity with config %+v", s, ai.Config)
+	log.Lvlf3("%s Creating new identity with data %+v", s, ai.Data)
 	ids := &Storage{
-		Latest: ai.Config,
+		Latest: ai.Data,
 	}
 	log.Lvl3("Creating Root-skipchain")
 	var cerr onet.ClientError
-	ids.Root, cerr = s.skipchain.CreateGenesis(ai.Roster, 10, 10,
+	ids.SCRoot, cerr = s.skipchain.CreateGenesis(ai.Roster, 10, 10,
 		[]skipchain.VerifierID{}, nil, nil)
 	if cerr != nil {
 		return nil, cerr
 	}
-	log.Lvl3("Creating Data-skipchain")
-	ids.Data, cerr = s.skipchain.CreateGenesis(ai.Roster, 10, 10,
-		[]skipchain.VerifierID{}, ai.Config, ids.Root.Hash)
+	log.Lvl3("Creating Data-skipchain", ai.Data)
+	ids.SCData, cerr = s.skipchain.CreateGenesis(ids.SCRoot.Roster, 10, 10,
+		VerificationIdentity, ai.Data, ids.SCRoot.Hash)
 	if cerr != nil {
 		return nil, cerr
 	}
 
-	roster := ids.Root.Roster
+	roster := ids.SCRoot.Roster
 	replies, err := s.propagateIdentity(roster, &PropagateIdentity{ids}, propagateTimeout)
 	if err != nil {
 		return nil, onet.NewClientErrorCode(ErrorOnet, err.Error())
 	}
 	if replies != len(roster.List) {
 		log.Warn("Did only get", replies, "out of", len(roster.List))
 	}
-	log.Lvlf2("New chain is\n%x", []byte(ids.Data.Hash))
+	log.Lvlf2("New chain is\n%x", []byte(ids.SCData.Hash))
 
 	return &CreateIdentityReply{
-		Root: ids.Root,
-		Data: ids.Data,
+		Root: ids.SCRoot,
+		Data: ids.SCData,
 	}, nil
 }
 
-// ConfigUpdate returns a new configuration update
-func (s *Service) ConfigUpdate(cu *ConfigUpdate) (network.Message, onet.ClientError) {
+// DataUpdate returns a new data-update
+func (s *Service) DataUpdate(cu *DataUpdate) (network.Message, onet.ClientError) {
+	// Check if there is something new on the skipchain - in case we've been
+	// offline
 	sid := s.getIdentityStorage(cu.ID)
 	if sid == nil {
 		return nil, onet.NewClientErrorCode(ErrorBlockMissing, "Didn't find Identity")
 	}
 	sid.Lock()
 	defer sid.Unlock()
-	log.Lvl3(s, "Sending config-update")
-	return &ConfigUpdateReply{
-		Config: sid.Latest,
+	reply, cerr := s.skipchain.GetUpdateChain(sid.SCRoot.Roster, sid.SCData.Hash)
+	if cerr != nil {
+		return nil, cerr
+	}
+	if len(reply.Update) > 1 {
+		log.Lvl3("Got new data")
+		// TODO: check that update-chain is legit
```

Comment:
I just verified, and indeed, there is a `// TODO - verify new config` in there ;)